lib/journal/src/concrete/compacting.rs

```diff
@@ -31,7 +32,7 @@ impl From<Range<u64>> for MemoryRange {
 }
 
 #[derive(Debug, Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Hash)]
-struct DescriptorLookup(u64);
+struct SubEventsLookup(u64);
```

Comment:
The name here was changed, but the type itself is still called `StateDscriptor`.
That's somewhat confusing.

Would this be better: `EventBatch` + `EventBatchId` ?